AK/bits/Math/x86.h

```diff
@@ -0,0 +1,416 @@
+/*
+ * Copyright (c) 2021, Leon Albrecht <leon2002.la@gmail.com>.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include "./common.h"
+
+#include <AK/Concepts.h>
+#include <AK/NumericLimits.h>
+#include <AK/StdLibExtraDetails.h>
+#include <AK/Types.h>
+
+namespace AK {
+
+namespace Details {
+template<size_t>
+constexpr size_t product_even();
+template<>
+constexpr size_t product_even<2>() { return 2; }
+template<size_t value>
+constexpr size_t product_even() { return value * product_even<value - 2>(); }
+
+template<size_t>
+constexpr size_t product_odd();
+template<>
+constexpr size_t product_odd<1>() { return 1; }
+template<size_t value>
+constexpr size_t product_odd() { return value * product_odd<value - 2>(); }
+}
+
+#define constexpr_STATE(function, args...)        \
```

Comment:
Is the owner-casing of "constexpr" in this macro intentional?

Also: The name "CONSTEXPR_STATE" seems historical, but is it still the most expressive name?

Reply:
Ohh that was some overeager use of search and replace
And its a macro so it should stay all caps, if you have a better name I can change it, this was what it was in the original implementation

Reply:
They say that naming is the hardest problem in computer science :) so I am afraid that I don't have much to offer here... TBH, my biggest gripe with that macro is that it contains a return statement so the execution can return from the method in a way that is a bit hidden from a casual reader of the code. But that is a larger issue, I presume.

Reply:
Let me retract my comment about my biggest gripe. That code is now local to x86.h, i.e., part of the special case that is the Intel architecture and hence not a general concern. Please disregard :)